src/shared/components/palette/dynamic-list/read-dynamic-list-field.component.spec.ts

```diff
@@ -0,0 +1,191 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { ReadDynamicListFieldComponent } from './read-dynamic-list-field.component';
+import { DebugElement } from '@angular/core';
+import { FieldType } from '../../../domain/definition/field-type.model';
+import { DynamicListPipe } from './dynamic-list.pipe';
+import { CaseField } from '../../../domain/definition/case-field.model';
+
+describe('ReadDynamicListFieldComponent', () => {
+
+  const VALUE = 'F';
+  const EXPECTED_LABEL = 'Female';
+  const FIELD_ID = 'ReadOnlyFieldId';
+  const FIELD_TYPE: FieldType = {
+    id: 'Gender',
+    type: 'FixedList',
+    fixed_list_items: [
+      {
+        code: 'M',
+        label: 'Male',
+        order: 1
+      },
+      {
+        code: VALUE,
+        label: EXPECTED_LABEL,
+        order: 2
+      },
+      {
+        code: 'O',
+        label: 'Other',
+        order: 3
+      }
+    ]
+  };
+
+  describe('Non-persistable readonly fixed list field', () => {
+    const ITEMS = [
+      {
+        code: 'M',
+        label: 'Male',
+        order: 1
+      },
+      {
+        code: VALUE,
+        label: EXPECTED_LABEL,
+        order: 2
+      },
+      {
+        code: 'O',
+        label: 'Other',
+        order: 3
+      }
+    ];
+
+    const CASE_FIELD: CaseField = Object.assign(new CaseField(), {
+      id: 'x',
+      label: 'X',
+      display_context: 'OPTIONAL',
+      field_type: FIELD_TYPE,
+      value: VALUE,
+    });
+
+    const EMPTY = '';
+
+    let fixture: ComponentFixture<ReadDynamicListFieldComponent>;
+    let component: ReadDynamicListFieldComponent;
+    let de: DebugElement;
+
+    beforeEach(async(() => {
+      TestBed
+        .configureTestingModule({
+          imports: [],
+          declarations: [
+            ReadDynamicListFieldComponent,
+            DynamicListPipe
+          ],
+          providers: []
+        })
+        .compileComponents();
+
+      fixture = TestBed.createComponent(ReadDynamicListFieldComponent);
+      component = fixture.componentInstance;
+
+      component.caseField = CASE_FIELD;
+
+      de = fixture.debugElement;
+      fixture.detectChanges();
+    }));
+
+    it('render label associated to the value provided', () => {
+      component.caseField.value = VALUE;
+      fixture.detectChanges();
+
+      expect(de.nativeElement.textContent).toEqual(EXPECTED_LABEL);
+    });
+
+    it('render undefined value as empty string', () => {
+      component.caseField.value = undefined;
+      fixture.detectChanges();
+
+      expect(de.nativeElement.textContent).toEqual(EMPTY);
+    });
+
+    it('render null value as empty string', () => {
+      component.caseField.value = null;
+      fixture.detectChanges();
+
+      expect(de.nativeElement.textContent).toEqual(EMPTY);
+    });
+  });
+
+  describe('ReadDynamicListFieldComponent for dynamiclist type', () => {
+    const VALUE_DYNAMIC_LIST = '{\n' +
```

Comment:
A string literal would be much easier to read and avoid the need for the `\n`s entirely.